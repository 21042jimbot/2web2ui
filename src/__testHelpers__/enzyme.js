export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
