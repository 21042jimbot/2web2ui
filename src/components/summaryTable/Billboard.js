import React from 'react';
import classNames from 'classnames';
import { Box, Table } from 'src/components/matchbox';
import { useHibana } from 'src/context/HibanaContext';
import styles from './SummaryTable.module.scss'; // TODO: Remove style import when OG theme is removed

function Billboard({ children, colSpan }) {
  const [state] = useHibana();
  const { isHibanaEnabled } = state;

  return (
    <tbody>
      <Table.Row>
        <Table.Cell className={classNames(!isHibanaEnabled && styles.Billboard)} colSpan={colSpan}>
          <Box height="220px">{children}</Box>
        </Table.Cell>
      </Table.Row>
    </tbody>
  );
}

export default Billboard;
