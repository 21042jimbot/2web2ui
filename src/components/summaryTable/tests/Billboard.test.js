import React from 'react';
import { shallow } from 'enzyme';
import Billboard from '../Billboard';

describe('Billboard', () => {
  it('renders signle cell table row', () => {
    const wrapper = shallow(
      <Billboard colSpan={7}>
        <h1>Test Example</h1>
      </Billboard>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
