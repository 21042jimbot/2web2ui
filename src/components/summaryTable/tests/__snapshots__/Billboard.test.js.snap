// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Billboard renders signle cell table row 1`] = `
<tbody>
  <Table.Row>
    <Table.Cell
      className="Billboard"
      colSpan={7}
    >
      <h1>
        Test Example
      </h1>
    </Table.Cell>
  </Table.Row>
</tbody>
`;
